### PR TITLE
[AR-224] index display name

### PR DESCRIPTION
**Pull request by lazysoundsystem**
Noting that there might be some changes with fixing the 'taxonomy' typo - at the moment it's using 'name' for taxonomy terms, which is wished for in the 'todo'.

Also, not sure if this is the right place to index the display name, or if we want to use 'display_name' as a standard and index those if they exist for other referenced entities.

### CODE REVIEW

**Comment by lazysoundsystem** [comment]
Confirming that fixing the typo caused the tests to fail:

```
GET http://docstore-travis-site/api/v1/documents/test-document-crud/1c0981a7-91ab-4ea3-be7e-aab08dae0a45

  Accept: "application/json"

  API-KEY: "abcd"


{"uuid":"1c0981a7-91ab-4ea3-be7e-aab08dae0a45","revision_id":2,"langcode":"en","published":true,"created":"2021-04-06T15:38:07+00:00","changed":"2021-04-06T15:38:07+00:00","author":"hid_123456789","private":true,"silk_country":{"uuid":"9f3a9614-7a2d-49b2-9fc8-57d90138352e","label":"Aruba"},"silk_my_id":42,"type":"test_doc_crud","type_uuid":"e2163259-bafe-4058-aca7-d039d66c1096","title":"Private","provider_uuid":"0a1fb2ab-bb06-466c-aca3-a199623da3e3"}


Data.silk_country.name expected string: "Aruba"  actual: (missing)  

--- FAIL: GET /documents/test-document-crud/{doc2} 

 silk_document_crud.md:495 - Data.silk_country.name doesn't match  

--- FAIL: silk (6.15s)

FAIL

The command "docker exec -it -u appuser -w /srv/www/tests -e HOST="http://docstore-travis-site" -e DRUSH="drush -v"  docstore-travis-site sh ./run.sh" exited with 1.
```

I made 'name' the label for taxonomy terms for now, but this needs more thought.

File: html/modules/custom/docstore/src/ResourceTrait.php [hunk 1]
```diff
@@ -549,8 +549,19 @@ public function prepareEntityEntityReferenceField(EntityReferenceFieldItemList $
...
-      if ($entity->getEntityTypeId() === 'taxonony_term') {
-        $label_key = 'label';
+      if ($entity->getEntityTypeId() === 'taxonomy_term') {
+        // @todo This was included when there was a typo in 'taxonomy_term',
+        // but it causes tests to fail, so specifying 'name' for now.
+        // $label_key = 'label';
+        $label_key = 'name';
+
+        // Add display label to local_coordination_groups.
+        if ($entity->vid->first()->getValue()['target_id'] === 'local_coordination_groups') {
```
Comment (orakili) [inline html/modules/custom/docstore/src/ResourceTrait.php:559]: Not a big fan of having special cases like that in code. There may be more use cases for a term display label on other vocabularies so I would suggest to simply check if it exists and not empty (`$entity->hasField('display_label') && !$entity->display_label->isEmpty()`) and then add it (`$entity->display_label->value`).

File: html/modules/custom/docstore/src/ResourceTrait.php [hunk 2]
```diff
@@ -549,8 +549,19 @@ public function prepareEntityEntityReferenceField(EntityReferenceFieldItemList $
       // @todo It would much easier if we were to use "name" for terms which is
       // the field name for the term label rather than "label" or if we were to
       // use "label" for all the entity types.
-      if ($entity->getEntityTypeId() === 'taxonony_term') {
-        $label_key = 'label';
+      if ($entity->getEntityTypeId() === 'taxonomy_term') {
+        // @todo This was included when there was a typo in 'taxonomy_term',
+        // but it causes tests to fail, so specifying 'name' for now.
```
Comment (orakili) [inline html/modules/custom/docstore/src/ResourceTrait.php:554]: `label` is the property name used when creating or retrieving terms so for consistency it should be 
"label" as well here and the tests modified. That may have an impact on the current AR (not anymore normally after https://github.com/UN-OCHA/assessmentregistry8-site/pull/233 is merged).

**Review by orakili — CHANGES_REQUESTED** [review]
Thanks for picking up the typo. Rather than keeping the wrong label key to please the tests I would prefer we make the returned data consistent (`label` is the label property for term resources, `name` is only internal) and modify the tests.

See other comments below.

**Comment by lazysoundsystem** [comment]
This is ready for another look. And tests updated to expect 'name' rather than 'label' for taxonomy terms.

(Might need the docstore_local_coordination_groups.php script run to create the display_label field.)

File: html/modules/custom/docstore/src/ResourceTrait.php [hunk 3]
```diff
@@ -570,6 +570,11 @@ public function prepareEntityEntityReferenceField(EntityReferenceFieldItemList $
         $item = array_merge($item, $extra_data);
       }
 
+      // Add display label if there is one.
+      if (isset($entity->display_label) && !$entity->display_label->isEmpty()) {
+        $item['display_label'] = $entity->display_label->first()->getValue()['value'];
```
Comment (orakili) [inline html/modules/custom/docstore/src/ResourceTrait.php:575]:
```suggestion
        $item['display_label'] = $entity->display_label->value;
```

**Review by orakili — CHANGES_REQUESTED** [review]
Please create a test for the `display_label` (see efforts to increase code coverage: https://github.com/UN-OCHA/docstore-site/pull/135).

**Review by attiks — APPROVED** [review]

**Pull request merged**